File: LeetCode/cpp/452.cpp
class Solution {
public:
    int findMinArrowShots(vector<pair<int, int>>& points) {
        sort(points.begin(), points.end(), [](pair<int, int>& p1, pair<int, int>& p2){
            return p1.first < p2.first;
        });
        if (points.size() == 0)
            return 0;
        int start = points[0].first, end = points[0].second, count = 0;
        for (int i = 1; i < points.size(); ++i) {
            if (max(points[i].first, start) <= min(points[i].second, end)) {
                start = max(points[i].first, start);
                end = min(points[i].second, end);
            }
            else {
                ++count;
                start = points[i].first;
                end = points[i].second;
            }
        }
        return count + 1;
    }
};

File: LeetCode/python3/452.py
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        points.sort(key=lambda x: x[0])
        if len(points) == 0:
            return 0
        start, end, count = points[0][0], points[0][1], 0
        for i in range(1, len(points)):
            if max(start, points[i][0]) <= min(end, points[i][1]):
                start = max(start, points[i][0])
                end = min(end, points[i][1])
            else:
                count += 1
                start = points[i][0]
                end = points[i][1]
        return count + 1
